exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverStatsExporter.java

```diff
@@ -365,6 +391,38 @@ static MetricServiceClient createMetricServiceClient(@Nullable Credentials crede
     if (credentials != null) {
       settingsBuilder.setCredentialsProvider(FixedCredentialsProvider.create(credentials));
     }
+
+    // We use createMetricDescriptor and createTimeSeries APIs in this exporter.
+    UnaryCallSettings.Builder<CreateMetricDescriptorRequest, MetricDescriptor>
+        createMetricDescriptorSettings = settingsBuilder.createMetricDescriptorSettings();
+    UnaryCallSettings.Builder<CreateTimeSeriesRequest, Empty> createTimeSeriesSettings =
+        settingsBuilder.createTimeSeriesSettings();
+    if (deadline != null) {
+      org.threeten.bp.Duration stackdriverDuration =
+          org.threeten.bp.Duration.ofMillis(deadline.toMillis());
+      createMetricDescriptorSettings.setSimpleTimeoutNoRetries(stackdriverDuration);
+      createTimeSeriesSettings.setSimpleTimeoutNoRetries(stackdriverDuration);
+    } else {
+      /*
```

Comment:
Similarly with tracing - iiuc, the only way to get the default retry settings is to explicitly set the deadline - will that negatively impact any customers, i.e. are there customers that have been relying on retries?

Reply:
Similar comment to https://github.com/census-instrumentation/opencensus-java/pull/1881#discussion_r281708276. I'll add a commit to allow users to use a custom `MetricClient` so that they can use custom settings.

Reply:
Sounds good.

Reply:
Added the new option in https://github.com/census-instrumentation/opencensus-java/pull/1883/commits/730e37ae91e4c4c054c025c9b85947691015e200, PTAL.